pkg/models/kubeconfig/kubeconfig.go

```diff
@@ -21,295 +21,210 @@ package kubeconfig
-		BasicConstraintsValid: true,
-		IsCA:                  info.IsCA,
-		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
-		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
-		EmailAddresses:        info.EmailAddress,
 	}
-}
 
-func generateCaAndKey(user, caPath, keyPath string) (string, string, error) {
-	crtInfo := CertInformation{CommonName: user, IsCA: false}
+	// create csr
+	k8sCSR, err = k8sClient.CertificatesV1beta1().CertificateSigningRequests().Create(k8sCSR)
 
-	crt, pri, err := Parse(caPath, keyPath)
 	if err != nil {
-		klog.Error(err)
-		return "", "", err
+		klog.Errorln(err)
+		return nil, err
 	}
-	cert, key, err := createCRT(crt, pri, crtInfo)
-	if err != nil {
-		klog.Error(err)
-		return "", "", err
+
+	k8sCSR.Status = certificatesv1beta1.CertificateSigningRequestStatus{
+		Conditions: []certificatesv1beta1.CertificateSigningRequestCondition{{
+			Type:    "Approved",
+			Reason:  "KubeSphereApprove",
+			Message: "This CSR was approved by KubeSphere certificate approve.",
+			LastUpdateTime: metav1.Time{
+				Time: time.Now(),
+			},
+		}},
 	}
 
-	base64Cert := base64.StdEncoding.EncodeToString(cert)
-	base64Key := base64.StdEncoding.EncodeToString(key)
-	return base64Cert, base64Key, nil
-}
+	// approve csr
+	k8sCSR, err = k8sClient.CertificatesV1beta1().CertificateSigningRequests().UpdateApproval(k8sCSR)
 
-func createKubeConfig(username string) (string, error) {
-	tmpKubeConfig := kubeConfig{ApiVersion: "v1", Kind: "Config"}
-	serverCa, err := ioutil.ReadFile(caPath)
 	if err != nil {
 		klog.Errorln(err)
-		return "", err
+		return nil, err
 	}
-	base64ServerCa := base64.StdEncoding.EncodeToString(serverCa)
-	tmpClusterInfo := clusterInfo{CertificateAuthorityData: base64ServerCa, Server: client.ClientSets().K8s().Master()}
-	tmpCluster := cluster{Cluster: tmpClusterInfo, Name: clusterName}
-	tmpKubeConfig.Clusters = append(tmpKubeConfig.Clusters, tmpCluster)
 
-	contextName := username + "@" + clusterName
-	tmpContext := contextObject{Context: contextInfo{User: username, Cluster: clusterName, NameSpace: defaultNamespace}, Name: contextName}
-	tmpKubeConfig.Contexts = append(tmpKubeConfig.Contexts, tmpContext)
-
-	cert, key, err := generateCaAndKey(username, caPath, keyPath)
+	// get client cert
+	k8sCSR, err = k8sClient.CertificatesV1beta1().CertificateSigningRequests().Get(csrName, metav1.GetOptions{})
 
 	if err != nil {
-		return "", err
+		klog.Errorln(err)
+		return nil, err
 	}
 
-	tmpUser := user{User: userInfo{CaData: cert, KeyData: key}, Name: username}
-	tmpKubeConfig.Users = append(tmpKubeConfig.Users, tmpUser)
-	tmpKubeConfig.CurrentContext = contextName
+	cert := k8sCSR.Status.Certificate
+
+	config := clientcmdapi.Config{
+		Kind:        "Config",
+		APIVersion:  "v1",
+		Preferences: clientcmdapi.Preferences{},
+		Clusters: map[string]*clientcmdapi.Cluster{"cluster.local": {
```

Comment:
do not hard code `cluster.local`, users may use a different domain

Reply:
`cluster.local` is default cluster name, not domain.